plugins/@grouparoo/postgres/src/lib/table-import/getPropertyValues.ts

```diff
@@ -21,55 +21,68 @@ export const getPropertyValues: GetPropertyValuesMethod = async ({
   primaryKeys,
 }) => {
   let responses: { [key: string]: { [column: string]: DataResponse[] } } = {};
-  let aggSelect = columnNames.map((col) => `"${col}"`).join(", ");
-  let orderBy = "";
-  let groupByColumns = [tablePrimaryKeyCol];
+  const columnList = columnNames.map((col) => `"${col}"`).join(", ");
+  const orderBys = [];
+  const groupByColumns = [];
+  let aggFunc = null;
 
   if (primaryKeys.length === 0) return responses;
 
   switch (aggregationMethod) {
     case AggregationMethod.Exact:
-      groupByColumns.push(...columnNames);
       if (sortColumn) {
-        orderBy = `"${sortColumn}" ASC`;
-        groupByColumns.push(sortColumn);
+        orderBys.push(`"${sortColumn}" ASC`);
       }
       break;
     case AggregationMethod.Average:
-      aggSelect = `COALESCE(AVG(${aggSelect}), 0) as ${aggSelect}`;
+      aggFunc = `COALESCE(AVG(${columnList}), 0) as ${columnList}`;
       break;
     case AggregationMethod.Count:
-      aggSelect = `COUNT(${aggSelect})::integer as ${aggSelect}`;
+      aggFunc = `COUNT(${columnList})::integer as ${columnList}`;
       break;
     case AggregationMethod.Sum:
-      aggSelect = `COALESCE(SUM(${aggSelect}), 0) as ${aggSelect}`;
+      aggFunc = `COALESCE(SUM(${columnList}), 0) as ${columnList}`;
       break;
     case AggregationMethod.Min:
-      aggSelect = `MIN(${aggSelect}) as ${aggSelect}`;
+      aggFunc = `MIN(${columnList}) as ${columnList}`;
       break;
     case AggregationMethod.Max:
-      aggSelect = `MAX(${aggSelect}) as ${aggSelect}`;
+      aggFunc = `MAX(${columnList}) as ${columnList}`;
       break;
     case AggregationMethod.MostRecentValue:
       if (!sortColumn) throw new Error("Sort Column is needed");
-      orderBy = `"${sortColumn}" DESC`;
-      groupByColumns.push(columnNames[0]);
-      groupByColumns.push(sortColumn);
+      orderBys.push(`"${sortColumn}" DESC`);
       break;
     case AggregationMethod.LeastRecentValue:
       if (!sortColumn) throw new Error("Sort Column is needed");
-      orderBy = `"${sortColumn}" ASC`;
-      groupByColumns.push(columnNames[0]);
-      groupByColumns.push(sortColumn);
+      orderBys.push(`"${sortColumn}" ASC`);
       break;
     default:
       throw new Error(`${aggregationMethod} is not a known aggregation method`);
   }
 
-  const params: Array<any> = [tableName];
-  let query = `SELECT ${aggSelect}, "${tablePrimaryKeyCol}" as __pk FROM %I WHERE`;
-  let addAnd = false;
+  const params: Array<any> = [];
+
+  let query = `SELECT`;
+
+  if (aggFunc || isArray) {
+    query += ` "${tablePrimaryKeyCol}" as __pk`;
+  } else {
+    query += ` DISTINCT ON (__pk) "${tablePrimaryKeyCol}" as __pk`;
```

Comment:
Postgres magic - https://www.geekytidbits.com/postgres-distinct-on/

I don't personally have any knowledge of if `DISTINCT ON` is faster or slower than GROUP BY.

Reply:
I'm fairly sure you can't just GROUP BY `user_id` 

For example, this query (most recent purchase dates) is invalid.

> SELECT "user_id" as __pk, "created_at" FROM demo.purchases WHERE  "state" = 'successful' AND  "user_id" IN ('1','2','3','4') GROUP BY "user_id" ORDER BY "created_at" DESC 

> ERROR:  column "purchases.created_at" must appear in the GROUP BY clause or be used in an aggregate function
LINE 1: SELECT "user_id" as __pk, "created_at" FROM demo.purchases W...

So you have to add `created_at` to the `GROUP BY`. This defeats the purpose as they end up not being grouped.
I believe this works in MySQL and, even better, makes the one that's selected the first in the list.

Reply:
It seems something like this is the way to do the above query without Postgres magic
> SELECT * FROM
>   (SELECT  "user_id" as __pk, "created_at", 
>               ROW_NUMBER() OVER (PARTITION BY "user_id" ORDER BY > "created_at" DESC) AS __rownum
>        FROM demo.purchases
>        WHERE  "state" = 'successful' AND  "user_id" IN ('1','2','3','4')
>        ORDER BY "created_at" DESC
>   ) AS __ranked
> WHERE __ranked.__rownum = 1;

Reply:
Ok, doing that in https://github.com/grouparoo/grouparoo/pull/2522